src/Index/Index.php

```diff
@@ -15,14 +15,31 @@ class Index implements ReadableIndex, \Serializable
     use EmitterTrait;
 
     /**
-     * An associative array that maps fully qualified symbol names to Definitions
+     * An associative array that maps namespaces to
+     *     an associative array that maps fully qualified symbol names
+     *     to global Definitions, e.g. :
+     *     [
+     *         'Psr\Log\LoggerInterface' => [
+     *             'Psr\Log\LoggerInterface->log()' => $definition,
+     *             'Psr\Log\LoggerInterface->debug()' => $definition,
```

Comment:
I'm not sure it worth it as we don't need this kind of representation for now. The actual representation is mainly used to get definitions of the resolved class directly (w/o having to parse its namespace).